### PR TITLE
tracefile-analyser.php revisions

### PR DESCRIPTION
**Pull request by Tinram**
- suppress warnings and errors for clearer output
- add minor whitespace changes to help readability
- add a shebang for when direct script execution is preferred
- add usage comment

### CODE REVIEW

File: contrib/tracefile-analyser.php [hunk 1]
```diff
@@ -182,7 +211,7 @@ private function parseLine( $line )
 
 				// collapse data onto functions array
 				$dTime   = $time   - $prevTime;
-				$dMemory = $memory - $prevMem;
+				$dMemory = (int) $memory - (int) $prevMem;
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: That shouldn't be necessary, as those values should always be integers already. How did this throw an error?

File: contrib/tracefile-analyser.php [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/php
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: This is not correct, it should be using ``/usr/bin/env``.

File: contrib/tracefile-analyser.php [hunk 3]
```diff
@@ -15,7 +16,20 @@
    +----------------------------------------------------------------------+
    | Authors: Derick Rethans <derick@xdebug.org>                          |
    +----------------------------------------------------------------------+
- */
+*/
+
+
+/*
+   php.ini additions for operation:
+      xdebug.profiler_enable = 1
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: This is not needed, as this file has nothing to do with the profiler.

File: contrib/tracefile-analyser.php [hunk 4]
```diff
@@ -15,7 +16,20 @@
...
+      debug.trace_format = 1
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: This is misspelled, as it misses the "x".

File: contrib/tracefile-analyser.php [hunk 5]
```diff
@@ -15,7 +16,20 @@
...
+      xdebug.collect_includes = 1   (optional)
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: I don't believe this is useful, as the analyser only reports function names, and this is for showing arguments to functions.

File: contrib/tracefile-analyser.php [hunk 6]
```diff
@@ -15,7 +16,20 @@
...
+*/
+
+
+error_reporting(0); // suppress notices and warnings which disrupt terminal output
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: Instead of hiding errors, I'd rather see them fixed.

File: contrib/tracefile-analyser.php [hunk 7]
```diff
@@ -240,4 +269,5 @@ function( $a, $b ) use ( $sortKey )
 		return $result;
 	}
 }
-?>
+
+?>
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: Please leave new lines at the end of the last line.

File: contrib/tracefile-analyser.php [hunk 8]
```diff
@@ -62,16 +78,25 @@
...
+		"%-{$maxLen}s %5d  %3.4f %8d  %3.4f %8d\n",
+		$name,
+		$f['calls'],
+		$f['time-inclusive'],
+		$f['memory-inclusive'],
+		$f['time-own'], $f['memory-own']
+	);
 }
 
+echo PHP_EOL;
```
Comment (derickr) [inline contrib/tracefile-analyser.php]: I'm using ``\n`` at the end of the ``printf`` already, this just adds an extra line before returning to the shell prompt, so I don't think this is needed.

**Comment by derickr** [comment]
I'm sorry to say, but literally nothing besides the trivial extra white space changes was actually correct. I'm therefore closing this PR.

**Comment by mlocati** [comment]
@Tinram hiding warnings is not the way to go, never. They are very useful to identify bugs for instance.

Think for instance at this example:
```php
$array = ['fooBar' => 1];
echo $array['foobar'];
```
Here we have the `Notice: Undefined index: foobar` warning, very useful

**Pull request closed without merging**